libs/wallet/src/connectors/json-rpc-connector.ts

```diff
@@ -91,116 +28,104 @@ export class JsonRpcConnector implements VegaConnector {
   url: string | null = null;
   token: string | null = null;
   reqId = 0;
+  client?: WalletClient;
 
   constructor() {
     const cfg = getConfig();
-    if (cfg) {
+
+    if (cfg && cfg.url) {
       this.token = cfg.token;
       this.url = cfg.url;
+      this.client = new WalletClient({
+        hostname: cfg.url,
+        token: cfg.token ?? undefined,
+      });
     }
   }
 
   async getChainId() {
-    const result = await this.request(Methods.GetChainId);
-    if ('error' in result) {
-      throw this.wrapError(result.error);
+    if (!this.client) {
+      throw new Error(
+        'No client found. The connector needs to be initialized with a url.'
+      );
     }
-    const parseResult = GetChainIdSchema.safeParse(result);
-    if (parseResult.success) {
-      return parseResult.data.result;
-    } else {
+    try {
+      const { result } = await this.client.GetChainId();
+      return result;
+    } catch (err) {
       throw ClientErrors.INVALID_RESPONSE;
     }
   }
 
   async connectWallet() {
-    const result = await this.request(Methods.ConnectWallet, {
-      hostname: window.location.host,
-    });
-    if ('error' in result) {
-      throw this.wrapError(result.error);
+    if (!this.client) {
+      throw new Error(
+        'No client found. The connector needs to be initialized with a url.'
+      );
     }
-    const parseResult = ConnectWalletSchema.safeParse(result);
 
-    if (parseResult.success) {
-      // store token and other config for eager connect and subsequent requests
+    try {
+      const { result } = await this.client.ConnectWallet();
       setConfig({
-        token: parseResult.data.result.token,
+        token: result.token,
         connector: 'jsonRpc',
         url: this.url,
       });
-      return parseResult.data.result;
-    } else {
+      return result;
+    } catch (err) {
       throw ClientErrors.INVALID_RESPONSE;
     }
   }
 
   // connect actually calling list_keys here, not to be confused with connect_wallet
   // which retrieves the session token
   async connect() {
-    const cfg = getConfig();
-    if (!cfg?.token) {
-      throw ClientErrors.NO_TOKEN;
-    }
-    const result = await this.request(Methods.ListKeys, {
-      token: cfg.token,
-    });
-    if ('error' in result) {
-      throw this.wrapError(result.error);
+    if (!this.client) {
+      throw new Error(
+        'No client found. The connector needs to be initialized with a url.'
+      );
     }
-    const parseResult = ListKeysSchema.safeParse(result);
-    if (parseResult.success) {
-      return parseResult.data.result.keys;
-    } else {
+
+    try {
+      const { result } = await this.client.ListKeys({});
```

Comment:
question (non blocking): Why the empty object here? Add a comment?

Reply:
it's a bug in the generator :( haven't figured out a way to be able to easily do this without bloating the template into some unreadable forever-entangled spaghetti. but since the pipeline is complaining on this, will see what i can do to clean this up, it's pretty annoying having to pass in empty objects as a consumer.

Reply:
fixed this in https://github.com/vegaprotocol/vegawallet-ui/pull/55 so now i was able to remove the empty object params